gym/spaces/box.py

```diff
@@ -78,16 +78,19 @@ def __init__(
 
         # determine shape if it isn't provided directly
         if shape is not None:
-            shape = tuple(shape)
-        elif not np.isscalar(low):
-            shape = low.shape  # type: ignore
-        elif not np.isscalar(high):
-            shape = high.shape  # type: ignore
+            shape = tuple(int(dim) for dim in shape)
+        elif isinstance(low, np.ndarray):
+            shape = low.shape
+        elif isinstance(high, np.ndarray):
+            shape = high.shape
+        elif np.issubdtype(type(low), np.integer) and np.issubdtype(
+            type(high), np.integer
+        ):
+            shape = (1,)
         else:
             raise ValueError(
-                "shape must be provided or inferred from the shapes of low or high"
+                f"Box shape is inferred from low and high, expect their type to be np.ndarray or np.integer, actual type low: {type(low)}, high: {type(high)}"
```

Comment:
Low and high don't have to be integers, they need to be arrays or individual numbers